Week07/lesson14/homework/responsive_articles_with_nav/css/styles.css

```diff
@@ -0,0 +1,229 @@
+/* Box Model Hack */
+* {
+ box-sizing: border-box;
+}
+
+/* Clearfix hack */
+.clearfix:before, .clearfix:after {
+  content: "";
+  display: table;
+}
+
+.clearfix:after{
+  clear: both;
+}
+
+body {
+  font-family: 'Lato', sans-serif;
+  font-size: 13px;
+  background-color: #F6F0E4;
+}
+
+/* Basic/Mobile Article Styling: */
+
+article {
+  padding: 20px;
+}
+
+h2 {
+  font-family: 'PT Serif', serif;
+  font-size: 2em;
+  margin: 20px 0;
+}
+
+p {
+  margin: 0;
+}
+
+img {
+  display: block;
+  width: 100%;
+}
+
+main {
+  margin: 0 auto;
+}
+
+header {
+  background-color: #7BAFA4;
+}
+
+nav {
+  width: 100%
+  padding-top: 60px;
+  padding-left: 10px;
+  padding-right: 10px;
+  clear: both;
+  overflow: hidden;
+}
+
+nav a:first-child{
+  border-top: 1px solid #F6F0E4;
+}
+
+nav a:last-child{
+  margin-bottom: 10px;
+}
+
+nav a {
+  border-bottom: 1px solid #F6F0E4;
+  text-decoration: none;
+  color: white;
+  display: block;
+  width: 100%;
+  padding-top: 20px;
+  padding-bottom: 20px;
+  padding-left: 10px;
+  margin: 0 auto;
+}
+
+nav a:focus, nav a:hover {
+  background-color: #488A7C;
+}
+
+.hamburger {
+  width: 40px;
+  height: 40px;
+  float: right;
+  margin-top: 10px;
+  margin-right: 10px;
+  margin-bottom: 10px;
+}
+
+.hamburger-line {
+  margin-top: 7px;
+  margin-bottom: 7px;
+  height: 3px;
+  background-color: white;
+}
+
+.hidden {
+  max-height: 0px;
+  transition: all .5s ease-in-out;
+  -webkit-transition: all .5s ease-in-out;
+   -moz-transition: all .5s ease-in-out;
+   -ms-transition: all .5s ease-in-out;
+   -o-transition: all .5s ease-in-out;
+}
+
+.show {
+  max-height: 400px;
+  transition: all .5s ease-in-out;
```

Comment:
The way `transition` works is that you set it up once and then use a class to manipulate whichever property it is you're changing/animating.

So in your case, you'll want:

```css
nav {
  width: 100%;
  max-height: 0px;
  padding: 60px 10px 0 10px; /* shorthand */
  clear: both;
  overflow: hidden;
  transition: max-height .5s ease-in-out; /* no need to transition all unless changing more than one thing */
}

.show {
  max-height: 400px;
}
```

... and you can get rid of `.hidden` entirely

After doing this you'll notice that while the animation works, the padding on `nav` is visible when it's closed:

![image](https://cloud.githubusercontent.com/assets/5670279/18257827/27fcc8fe-73c2-11e6-9244-5eda529a3e07.png)

... this is when I make my next point: padding and animation don't play nice together. Better to have the parent animate and have the child add the padding. So instead of:

```html
<nav>
  <a href="#">Tea & Coffee</a>
  <a href="#">Biscuits</a>
  <a href="#">Wines & Spirits</a>
  <a href="#">Homeware</a>
</nav>
```

go for:

```html
<nav>
  <div class="navlist">
    <a href="#">Tea & Coffee</a>
    <a href="#">Biscuits</a>
    <a href="#">Wines & Spirits</a>
    <a href="#">Homeware</a>
  </div>
</nav>
```

... and then animate the `nav` and pad the `.navlist`